sike/sike.go

```diff
@@ -0,0 +1,33 @@
+// Package sike contains all the required interfaces for running an instance of
+// Supersingular Isogeny Key Encapsulation (SIKE) mechanism between two parties.
+// This package provide all the required APIs and generic data-types for SIKE 
+// implementation without any specific underlying parameter.
+// Author: Amir Jalali		ajalali2016@fau.edu
+package sike
+
+import (
+	"io"
+)
+
+// SecretKey type for the key-encapsulation mechanism.
+type SecretKey interface {}
+
+// PublicKey type for the key-encapsulation mechanism.
+type PublicKey interface {}
+
+// Ciphertext type for the key-encapsulation mechanism.
+type Ciphertext interface {}
+
+// SIKE interface contains all the required APIs for running an instance of SIKE.
+type SIKE interface {
```

Comment:
Improvement: it would be worth having functions to import/export keys from/to hex